src/fswAlgorithms/attGuidance/oneAxisSolarArrayPoint/oneAxisSolarArrayPoint.h

```diff
@@ -0,0 +1,97 @@
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef _ONE_AXIS_SOLAR_ARRAY_POINT_
+#define _ONE_AXIS_SOLAR_ARRAY_POINT_
+
+#include <stdint.h>
+#include "architecture/utilities/bskLogging.h"
+#include "cMsgCInterface/AttRefMsg_C.h"
+#include "cMsgCInterface/BodyHeadingMsg_C.h"
+#include "cMsgCInterface/InertialHeadingMsg_C.h"
+#include "cMsgCInterface/NavTransMsg_C.h"
+#include "cMsgCInterface/EphemerisMsg_C.h"
+#include "cMsgCInterface/NavAttMsg_C.h"
+
+#define EPS 1e-12
+
+enum alignmentPriority{
+    prioritizeAxisAlignment = 0,
+    prioritizeSolarArrayAlignment = 1
+};
+
+enum bodyAxisInput{
+    inputBodyHeadingParameter = 0,
+    inputBodyHeadingMsg = 1
+};
+
+enum inertialAxisInput{
+    inputInertialHeadingParameter = 0,
+    inputInertialHeadingMsg = 1,
+    inputEphemerisMsg = 2
+};
+
+/*! @brief Top level structure for the sub-module routines. */
+typedef struct {
+
+    /* declare these quantities that always must be specified as flight software parameters */
+    double a1Hat_B[3];                           //!< arrays axis direction in B frame
+    int alignmentPriority;                       //!< flag to indicate which constraint must be prioritized
+
+    /* declare these optional quantities */
+    double h1Hat_B[3];                           //!< main heading in B frame coordinates
+    double h2Hat_B[3];                           //!< secondary heading in B frame coordinates
+    double hHat_N[3];                            //!< main heading in N frame coordinates
+    double a2Hat_B[3];                           //!< body frame heading that should remain as close as possible to Sun heading
+
+    /* declare these internal variables that are used by the module and should not be declared by the user */
+    int      bodyAxisInput;                      //!< flag variable to determine how the body axis input is specified 
+    int      inertialAxisInput;                  //!< flag variable to determine how the inertial axis input is specified 
+    int      count;                              //!< count variable used in the finite difference logic
+    uint64_t T1NanoSeconds;                      //!< callTime one update step prior
+    uint64_t T2NanoSeconds;                      //!< callTime two update steps prior
+    double   sigma_RN_1[3];                      //!< reference attitude one update step prior
+    double   sigma_RN_2[3];                      //!< reference attitude two update steps prior
+
+    /* declare module IO interfaces */
```

Comment:
:small_red_triangle_down: Remove this comment.

Reply:
Done